src/field/enemy_party.c

```diff
@@ -499,6 +503,13 @@ BOOL LONG_CALL AddWildPartyPokemon(int inTarget, EncounterInfo *encounterInfo, s
 
     WildMonSetRandomHeldItem(encounterPartyPokemon, encounterBattleParam->fight_type, range);
 
+    // In the wild, Super Size Pumpkaboo will always hold a Miracle Seed, whereas other sizes will never hold an item in the wild.
+#ifdef IMPLEMENT_DYNAMIC_WILD_SPECIES_FORMS
+    if ((species == SPECIES_PUMPKABOO || species == SPECIES_GOURGEIST) && GetMonData(encounterPartyPokemon, MON_DATA_FORM, NULL) == 3) {
+        SetMonData(encounterPartyPokemon, MON_DATA_HELD_ITEM, &miracleSeedItemNo);
+    }
+#endif
+
```

Comment:
i'm just going to fix this properly by setting the held item factoring in the form

Reply:
yea it already works lol
![image](https://github.com/user-attachments/assets/9091ee0a-3026-43e6-9d42-d8dbdd1e0d8c)